File: hdf-converters/converters/hdf-to-oscal-poam/go/anchor_test.go
package hdftooscalpoam

import (
	"testing"

	shared "github.com/mitre/hdf-libs/hdf-converters/v3/shared/go"
	fixtures "github.com/mitre/hdf-libs/hdf-fixtures"
	"github.com/stretchr/testify/require"
)

// TestConvertHDFToOSCALPOAM_OutputCountAnchor is the export-side ground-truth
// anchor: hdf-to-oscal-poam emits one poam-item per HDF override, so the emitted
// poam-item count must equal the override count derived independently from the
// HDF Amendments input (top-level overrides[], a raw JSON walk — not the
// converter's parser). The MultiCVE fixture has 3 overrides.
func TestConvertHDFToOSCALPOAM_OutputCountAnchor(t *testing.T) {
	input := fixtures.Amendments.MultiCVE
	want := shared.CountHDFOverrides(t, input)
	require.Greater(t, want, 1, "fixture must have multiple overrides for a meaningful anchor")

	out, err := ConvertHDFToOSCALPOAM(input, "1.0.0")
	require.NoError(t, err)

	got := shared.CountJSONItemsUnderKey(t, out, "poam-items")
	require.Equal(t, want, got, "one OSCAL poam-item per HDF override")
}

File: hdf-converters/converters/hdf-to-oscal-poam/typescript/anchor.test.ts
import { describe, it, expect } from 'vitest';
import { amendments } from '@mitre/hdf-fixtures';
import { countHdfOverrides, countJsonItemsUnderKey } from '../../../shared/typescript/anchor.js';
import { convertHdfToOscalPoam } from './converter.js';

// Export-side ground-truth anchor: one OSCAL poam-item per HDF override. Mirrors Go.
describe('hdf-to-oscal-poam output-count anchor', () => {
  it('emits one poam-item per HDF override', async () => {
    const input = amendments.multiCve.read();
    const want = countHdfOverrides(input);
    expect(want).toBeGreaterThan(1);
    const got = countJsonItemsUnderKey(await convertHdfToOscalPoam(input), 'poam-items');
    expect(got, 'one OSCAL poam-item per HDF override').toBe(want);
  });
});
